lib/matplotlib/tests/test_image.py

```diff
@@ -1218,3 +1219,35 @@ def test_huge_range_log(fig_test, fig_ref):
     ax = fig_ref.subplots()
     im = ax.imshow(data, norm=colors.Normalize(vmin=100, vmax=data.max()),
                    interpolation='nearest', cmap=cmap)
+
+
+@check_figures_equal()
+def test_spy_box(fig_test, fig_ref):
+    # setting up reference and test
+    ax_test = fig_test.subplots(1, 3)
+    ax_ref = fig_ref.subplots(1, 3)
+
+    i = 0
+
+    plot_data = [[1, 1], [1, 1], ], [[0, 0], [0, 0], ], [[0, 1], [1, 0], ]
+
+    for z, title in zip((plot_data), ["ones", "zeros", "mixed"]):
+        print(i, z, title)
```

Comment:
```suggestion
    plot_data = (
        [[1, 1], [1, 1]],
        [[0, 0], [0, 0]],
        [[0, 1], [1, 0]],
    )
    plot_titles = ["ones", "zeros", "mixed"]

    for i, (z, title) in enumerate(zip(plot_data, plot_titles)):
```
and delete the `i += 1` at the end.